demo/sections/components/RunGraphDemo.vue

```diff
@@ -1,45 +1,35 @@
 <template>
   <p-layout-default class="run-graph-demo">
     <RunGraph :config="config" class="run-graph-demo__graph" />
-    <!-- <p-number-input v-model="zoom" /> -->
   </p-layout-default>
 </template>
 
-<!-- eslint-disable camelcase -->
 <script lang="ts" setup>
-  import { randomId } from '@prefecthq/prefect-design'
-  import { endOfHour, startOfHour, startOfDay, endOfDay } from 'date-fns'
-  // import { ref } from 'vue'
+  import { parseISO, isValid } from 'date-fns'
   import RunGraph from '@/components/RunGraph.vue'
+  import json from '@/demo/data/graph-small.json'
   import { RunGraphConfig, RunGraphData } from '@/models'
 
-  // const zoom = ref(0)
-  const now = new Date()
-  const id = randomId()
-  const dummy: RunGraphData = {
-    start_time: startOfDay(now),
-    end_time: endOfDay(now),
-    root_node_ids: [id],
-    nodes: new Map([
-      [
-        id
-        , {
-          start_time: startOfHour(now),
-          end_time: endOfHour(now),
-          kind: 'task-run',
-          state_name: 'Completed',
-          label: 'bar',
-          child_ids: [],
-          parent_ids: [],
-          id,
-        },
-      ],
-    ]),
+  // quick and dirty way to convert the iso strings into actual dates.
+  function reviver(key: string, value: any): any {
```

Comment:
Yes this is just for the demo. 

Thats an interesting thought. Might be worth just keeping the date strings from the api. But generally its much easier to work with real date objects. But might be more efficient to convert to dates when needed and just leave the strings. I'll think about that

Reply:
Yeah that's mostly what I was thinking - since this package is primarily dealing with rendering the graph and not any auxiliary data, the fewer bulk transformations, the better. Otherwise if we're doing this anyway there's not much standing in the way of a full mapping layer for snake-case => camelcase and whatever else

Reply:
IMO something like this is a much lighter and more performant touch than doing a full map of the whole response. My though was to accept the json from the api and use a reviver like this for the parsing. That way the JSON.parse is the only transformation that is happening. Rather than mapping the already parsed json and doing conversions on every property. 

We've seen how slow that can be for the existing graph api.